saged/download_recount3.R

```diff
@@ -0,0 +1,239 @@
+# This script downloads and preprocesses gene-level data and metadata from
+# all samples in recount3
+#
+# Output:
+# A tsv file containing an array of gene expression data, and a metadata file
+
+# R version used: 4.1.0
+
+# Some conda issue causes the installation of GenomeInfoDbData to fail silently, preventing
+# recount3 from being installed. I recommend running this script from a base r installation 
+# rather than a conda env
+
+#install.packages("BiocManager", repos = "https://cran.r-project.org", dependencies=TRUE)
+#install.packages("dplyr", repos = "https://cran.r-project.org", dependencies=TRUE)
+#install.packages("tibble", repos = "https://cran.r-project.org", dependencies=TRUE)
+#install.packages("tidyr", repos = "https://cran.r-project.org", dependencies=TRUE)
+#BiocManager::install()
+#BiocManager::install("recount3")
+
+library(recount3)
+library(dplyr)
+library(tibble)
+library(tidyr)
+
+`%>%` <- dplyr::`%>%`
+
+getMetadata <- function(project, project_home, metadata_df) {
+  # Append the metadata for a project to a dataframe
```

Comment:
That's a good idea! For my use case I actually prefer the data and metadata to be separate from each other since I'll load them at different times. I think I'm fine with writing them to tsvs to be loaded in python, but I'll definitely remember `SummarizedExperiments` for future projects